internal/pkg/nvmlprovider/provider.go

```diff
@@ -88,22 +88,22 @@ func GetMIGDeviceInfoByID(uuid string) (*MIGDeviceInfo, error) {
 
 	tokens := strings.SplitN(uuid, "-", 2)
 	if len(tokens) != 2 || tokens[0] != "MIG" {
-		return nil, fmt.Errorf("Unable to parse UUID as MIG device")
+		return nil, fmt.Errorf("unable to parse UUID '%s' as MIG device", uuid)
```

Comment:
This is an optional change to enhance message, if required this can be reverted.